pkg/apis/pipeline/v1alpha1/git_resource_test.go

```diff
@@ -0,0 +1,159 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1_test
+
+import (
+	"testing"
+
+	"github.com/tektoncd/pipeline/test/names"
+
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	tb "github.com/tektoncd/pipeline/test/builder"
+)
+
+func Test_Invalid_NewGitResource(t *testing.T) {
+	cases := []struct {
+		desc             string
+		pipelineResource *v1alpha1.PipelineResource
+	}{
+		{
+			desc:             "Wrong resource type",
+			pipelineResource: tb.PipelineResource("git-resource", "default", tb.PipelineResourceSpec(v1alpha1.PipelineResourceTypeGCS)),
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.desc, func(t *testing.T) {
+			_, err := v1alpha1.NewGitResource(tc.pipelineResource)
+			if err == nil {
+				t.Error("Expected error creating Git resource")
+			}
+		})
+	}
+}
+
+func Test_Valid_NewGitResource(t *testing.T) {
+	cases := []struct {
+		desc             string
+		pipelineResource *v1alpha1.PipelineResource
+		want             *v1alpha1.GitResource
+	}{
+		{
+			desc: "With Revision",
+			pipelineResource: tb.PipelineResource("git-resource", "default",
+				tb.PipelineResourceSpec(v1alpha1.PipelineResourceTypeGit,
+					tb.PipelineResourceSpecParam("URL", "git@github.com:test/test.git"),
+					tb.PipelineResourceSpecParam("Revision", "test"),
+				),
+			),
+			want: &v1alpha1.GitResource{
+				Name:     "git-resource",
+				Type:     v1alpha1.PipelineResourceTypeGit,
+				URL:      "git@github.com:test/test.git",
+				Revision: "test",
+			},
+		},
+		{
+			desc: "Without Revision",
+			pipelineResource: tb.PipelineResource("git-resource", "default",
+				tb.PipelineResourceSpec(v1alpha1.PipelineResourceTypeGit,
+					tb.PipelineResourceSpecParam("URL", "git@github.com:test/test.git"),
+				),
+			),
+			want: &v1alpha1.GitResource{
+				Name:     "git-resource",
+				Type:     v1alpha1.PipelineResourceTypeGit,
+				URL:      "git@github.com:test/test.git",
+				Revision: "master",
+			},
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.desc, func(t *testing.T) {
+			got, err := v1alpha1.NewGitResource(tc.pipelineResource)
+			if err != nil {
+				t.Fatalf("Unexpected error creating Git resource: %s", err)
+			}
+
+			if diff := cmp.Diff(tc.want, got); diff != "" {
+				t.Errorf("Mismatch of Git resource: %s", diff)
+			}
+		})
+	}
+}
+
+func Test_GitResource_Replacements(t *testing.T) {
+	r := &v1alpha1.GitResource{
+		Name:       "git-resource",
+		Type:       v1alpha1.PipelineResourceTypeGit,
+		URL:        "git@github.com:test/test.git",
+		Revision:   "master",
+		TargetPath: "/test/test",
+	}
+
+	want := map[string]string{
+		"name":     "git-resource",
+		"type":     string(v1alpha1.PipelineResourceTypeGit),
+		"url":      "git@github.com:test/test.git",
+		"revision": "master",
+		"path":     "/test/test",
```

Comment:
Another test case here would be testing the `path` replacement when `TargetPath` is not specified. (Though we might be changing how that works based on #1115)

Reply:
I think we decided to keep that around, so it should be safe to add.